internal/auth/credentials.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"fmt"
+
+	"cloud.google.com/go/auth"
+	"cloud.google.com/go/auth/credentials"
+	"cloud.google.com/go/storage"
+)
+
+var detectCredentials = credentials.DetectDefault
+
+const scope = storage.ScopeFullControl
+
+// GetCredentials detects default Google Cloud credentials.
+//
+// It prioritizes a service account key file if `keyFile` is provided. If `keyFile` is
+// empty, it automatically attempts to detect Application Default Credentials (ADC)
+// and checks the metadata server for credentials.
+//
+// This method requests `storagev1.DevstorageFullControlScope` for Google Cloud Storage access.
```

Comment:
why does it requests for this scope?

Reply:
The function requests storagev1.DevstorageFullControlScope to ensure the most comprehensive permissions for GCS. This allows subsequent operations using these credentials to have full read, write, and administrative control over GCS resources. 

We've maintained this [scope](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/auth/auth.go#L95) from our older authentication method to avoid breaking any existing scenarios.

Reply:
Thanks for the clarification.